philander/stc311.py

```diff
@@ -0,0 +1,665 @@
+                    ret = self._transferChangeRate(data)
+                else:
+                    ret = SOCChangeRate.invalid
+            else:
+                ret = SOCChangeRate.invalid
+        return ret
+
+    def getBatteryTemperature(self):
+        # This device does not measure any temperature data
+        return Temperature.invalid
+
+    def getChipTemperature(self):
+        """Retrieve the current temperature of the chip.
+
+        :return: Temperature value.
+        :rtype: Temperature
+        """
+        data, err = SerialBusDevice.readByteRegister(self, self.REGISTER.REG_TEMPERATURE)
+        if err == ErrorCode.errOk:
+            ret = self._transferTemperature(data)
+        else:
+            ret = Temperature.invalid
+        return ret
+
+    # Interruptable API implementation
+
+    def registerInterruptHandler(self, onEvent=Event.evtInt1, callerFeedBack=None, handler=None):
+        if handler is not None:  # Enable; from app (=sink) to hardware (=source)
+            self.pinInt.registerInterruptHandler(onEvent, callerFeedBack, handler)
+            err = self.pinInt.enableInterrupt()
+            if err == ErrorCode.errOk:
+                data, err = SerialBusDevice.readByteRegister(self, self.REGISTER.REG_MODE)
+                if err == ErrorCode.errOk:
+                    data |= self.REGISTER.MODE_ALM_ENA
+                    err = SerialBusDevice.writeByteRegister(self, self.REGISTER.REG_MODE, data)
+                if err == ErrorCode.errOk:  # check if there already is an interrupt present
+                    data, err = SerialBusDevice.readByteRegister(self, self.REGISTER.REG_CTRL)
+                    if data & self.REGISTER.CTRL_IO0DATA:
+                        handler(Event.evtInt1, callerFeedBack)
+                else:
+                    self.disableInterrupt()
+            else:
+                err = ErrorCode.errInvalidParameter  # TODO: is this the right error code?
+        else:  # Disable; from hardware to app.
+            data, err = SerialBusDevice.readByteRegister(self, self.REGISTER.REG_MODE)
+            if err == ErrorCode.errOk:
+                data &= ~self.REGISTER.MODE_ALM_ENA  # TODO: ModeValues class need to be adjusted to work with all binary operations as expected
+                err = SerialBusDevice.writeByteRegister(self, self.REGISTER.REG_MODE, data)
+            self.disableInterrupt()
+        return err
+
+    def enableInterrupt(self):
+        # err = GPIO.enableInterrupt(self, self.REGISTER.CONFIG_GASGAUGE_0_GPIO_ALARM)  # TODO: implement GPIO interrupt, check if it does infact return an error (not sure)
+        # if err == ErrorCode.errOk:  # TODO: do we really want these lines to override this error, and not just return the given error?
+        #     ret = ErrorCode.errOk
+        # else:
+        #     ret = ErrorCode.errInvalidParameter
+        # return ret
+        return ErrorCode.errOk  # TODO: why doesn't this function do anything (see max77960)
```

Comment:
It should enable the interrupt at the GPIO pin, if existent.

Reply:
why does max77960 never call pinInt.enableInterrupt?

Reply:
Right, this is actually missing in the MAX77960 implementation. Still, it should be done (some day), here.